lib/autoupdate/agent/process.go

```diff
@@ -109,33 +120,69 @@ func (s SystemdService) Reload(ctx context.Context) error {
 	default:
 		s.Log.InfoContext(ctx, "Gracefully reloaded.", unitKey, s.ServiceName)
 	}
+	// monitor logs all relevant errors, so we filter for a few outcomes
+	err = s.monitor(ctx, initPID)
+	if errors.Is(err, context.DeadlineExceeded) ||
+		errors.Is(err, context.Canceled) {
+		return trace.Wrap(err)
+	}
+	if err != nil {
+		return trace.Errorf("failed to monitor process")
+	}
+	return nil
+}
+
+// monitor for a started, healthy process.
+// monitor logs all errors that should be displayed to the user.
+func (s SystemdService) monitor(ctx context.Context, initPID int) error {
+	ctx, cancel := context.WithTimeout(ctx, monitorTimeout)
+	defer cancel()
+	tickC := time.NewTicker(monitorInterval).C
```

Comment:
Is this intended to be backported to v16? We're using go 1.22 in branch/v16, which means that this construction will leak the ticker.